docs/DeveloperGuide.md

```diff
@@ -8,8 +8,237 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### EasInternship (main class)
 
-## Product scope
+1. [Design](#design)
+   1. [Architecture](#architecture)
+   2. [UI Component](#ui-component)
+   3. [Logic Component](#logic-component)
+   4. [Model Component](#model-component)
+   5. [Storage Component](#storage-component)
+2. [Implementation](#implementation)
+   1. [Parsing Input](#parsing-input)
+   2. [Command Execution](#command-execution)
+   3. [Task Management](#task-management) 
+   4. [AddCommand Implementation](#addcommand-implementation)
+   5. [DeleteCommand Implementation](#deletecommand-implementation)
+   6. [SortCommand Implementation](#sortcommand-implementation)
+   7. [UpdateCommand Implementation](#updatecommand-implementation)
+
+---
+
+## 1. Design
+
+### 1.1 Architecture
+The architecture of EasInternship is designed to follow the MVC (Model-View-Controller) pattern to facilitate separation of concerns, modularity, and maintainability.
+
+- **View (UI)**: Responsible for interacting with the user by displaying output and reading input.
+- **Controller (Command and Parser)**: Responsible for parsing user input and invoking the appropriate commands.
+- **Model (InternshipList)**: Manages the state of the application, including the list of internships and tasks.
+- **Storage**: Responsible for loading and saving data from and to the disk.
+
```

Comment:
I feel that an architecture diagram help could be a very helpful diagram to add here to give a brief overview of how your different classes interact with each other